spacy/ml/tb_framework.pyx

```diff
@@ -127,72 +139,83 @@ def init(
     Wl = ops.alloc2f(nH * nP, nF * nI)
     bl = ops.alloc1f(nH * nP)
     padl = ops.alloc1f(nI)
-    Wu = ops.alloc2f(nO, nH)
-    bu = ops.alloc1f(nO)
-    Wu = zero_init(ops, Wu.shape)
     # Wl = zero_init(ops, Wl.shape)
     Wl = glorot_uniform_init(ops, Wl.shape)
     padl = uniform_init(ops, padl.shape)  # type: ignore
     # TODO: Experiment with whether better to initialize upper_W
     model.set_param("lower_W", Wl)
     model.set_param("lower_b", bl)
     model.set_param("lower_pad", padl)
-    model.set_param("upper_W", Wu)
-    model.set_param("upper_b", bu)
     # model = _lsuv_init(model)
     return model
 
-def forward(model, docs_moves: Tuple[List[Doc], TransitionSystem], is_train: bool):
+InWithoutActions = Tuple[List[Doc], TransitionSystem]
+InWithActions = Tuple[List[Doc], TransitionSystem, List[Ints1d]]
+InT = TypeVar("InT", InWithoutActions, InWithActions)
+
+def forward(model, docs_moves: InT, is_train: bool):
+    if len(docs_moves) == 2:
+        docs, moves = docs_moves
+        actions = None
+    else:
+        docs, moves, actions = docs_moves
+
     beam_width = model.attrs["beam_width"]
     lower_pad = model.get_param("lower_pad")
     tok2vec = model.get_ref("tok2vec")
 
-    docs, moves = docs_moves
     states = moves.init_batch(docs)
     tokvecs, backprop_tok2vec = tok2vec(docs, is_train)
     tokvecs = model.ops.xp.vstack((tokvecs, lower_pad))
     feats, backprop_feats = _forward_precomputable_affine(model, tokvecs, is_train)
     seen_mask = _get_seen_mask(model)
 
+    # Fixme: support actions in forward_cpu
     if beam_width == 1 and not is_train and isinstance(model.ops, NumpyOps):
-        return _forward_greedy_cpu(model, moves, states, feats, seen_mask)
+        return _forward_greedy_cpu(model, moves, states, feats, seen_mask, actions=actions)
     else:
-        return _forward_fallback(model, moves, states, tokvecs, backprop_tok2vec, feats, backprop_feats, seen_mask, is_train)
+        return _forward_fallback(model, moves, states, tokvecs, backprop_tok2vec, feats, backprop_feats, seen_mask, is_train, actions)
```

Comment:
```suggestion
        return _forward_fallback(model, moves, states, tokvecs, backprop_tok2vec, feats, backprop_feats, seen_mask, is_train, actions=actions)
```
(for symmetry but more importantly I think it's good practice to use the `keyword=` style for optional arguments)